### PR TITLE
Authorization for deleting posts and comments from owner and admin user.

### PR DESCRIPTION
**Pull request by NelsonAraujo93**
## Hello, Reviewers 🎯

*To succeed in the review we modify the user role inside the rails console, to achieve this follow this instructions:*
1. create an user in the sing_up option.
2. confirm mail.
3. acces to rails consoles by runing rails c.
4. access to first user or the user you want to be the admin.
5. modify the user adding role = 'admin' and save. ( Ex: admin= User.first;  admin.role='admin';  admin.save )

**In this pull request @kensteph and i followed the next steps:**
1. We install cancancan gem.
6. We set up a new migration to add role column to user table.
7. We run the Ability command to create authorization rules to users.
8. We create the delete options inside partials post and comments.
9. We add detroy routes to comments and posts.
10. We fix linters.

### CODE REVIEW

File: app/controllers/users/posts_controller.rb [hunk 1]
```diff
@@ -27,6 +27,16 @@ def create
...
+  def destroy
+    @post = Post.find(params[:id])
+    @user = User.find(params[:user_id])
+    @post.destroy
+    @user.posts_counter -= 1
+    @user.save
+    flash[:notice] = "Post #{@post.id} was successfully deleted for user : #{@user.name}!"
+    redirect_to "/users/#{params[:user_id]}/posts"
+  end
+
```
Comment (AnselemOdims) [inline app/controllers/users/posts_controller.rb:40]: - [x] Good job working on the destroy method, but you missed out a key functionality which is to ensure that the Post Controller and also the Comment Controller uses CanCanCan for Post authorization. This is to ensure that a user has permissions to perform certain operations.

Hint:
`load_and_authorize_resource`

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @team, 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/controllers/users/posts_controller.rb (hunk 1 above)
Comment (NelsonAraujo93) [reply app/controllers/users/posts_controller.rb:40]: Thank you so much for the feedback.

**Review by BertrandConxy — APPROVED** [review]
Hi @NelsonAraujo93 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💪 
- No linter errors
- project requirements are met
- project works fine
- good PR description
- correct flow is used

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**